### ISSUE TITLE
Merging Person records can cause confusion in Group Membership

### ISSUE THREAD

**Issue by cabal95** [SparkDevNetwork/Rock#2809]
### Prerequisites
* [X] Put an X between the brackets on this line if you have done all of the following:
    * Can you reproduce the problem on a fresh install or the [demo site](http://rock.rocksolidchurchdemo.com/)?
    * Did you include your Rock version number and [client culture](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?
    * Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ASparkDevNetwork%2FSlack) to see if your bug or enhancement is already reported?

### TL;DR

Basically, if a person has two records and both records have a GroupMember record in the same group (the older one being inactive and the newer one being active), after the merge the person will have only a single GroupMember record in that group and it can be inactive (maybe always will be?), which I think is unexpected.

### Description

Exact steps to reproduce the issue are below. I don't know what the proper fix is. Here is our full use-case where we ran into this.

In our High School small groups, we re-use the groups each semester, assuming the same leader is coming back. At the start of each semester we bulk inactivate all members of the groups. The way we register people for groups is by using an Event Registration. I'm not sure of all the details but I believe one of the registration questions lets them indicate which leader they would like to have.

So when "John Doe" student attended "Will Smith's" small group last semester. The semester has ended and we bulk inactivated all the members. Now "John Doe" wants to attend "Will Smith's" group again, cuz he's just that dope.

Now, since this is done through Group Registration a new record is often created. I think most often because the e-mail address we have on file is the parent's e-mail but they enter the student's e-mail at registration time. For whatever reason, we have a second "John Doe" record. The staff person then goes in to use the Group Placement feature and places them in "Will Smith's" group. So now there are two "John Doe"s that are members of the group. One is an inactive GroupMember record and is older. The newer one is active.

Finally, a few days later our data quality people get around to merging these two "John Doe" records. This results in, apparently, the oldest GroupMember record taking precedence which results in "John Doe" suddenly being inactive in the group.

### Steps to Reproduce

1. Create a new group called "Test Group"
2. Create new person called "Test Person" as a Male (person A)
3. Add person A to the Test Group with an Inactive member status
4. Create new person called "Test Person" as a Female (person B)
5. Add person B to the Test Group with an Active member status
6. Merge both "Test Person" A and B together and choose Male as the merge choice
7. Notice membership in Test Group has been reduced to 1 record and it is inactive.

**Expected behavior:** 

Frankly, I'm not sure. There are I suppose a few options.

1. Leave both GroupMember records intact, though this might cause issues since the default web.config setting is to not allow duplicate group members.
2. Ask in the UI when "conflicting" group membership is found.
3. Merge the GroupMember records (i.e. keep oldest DateTimeAdded and Active status takes priority, but this could be an issue because there are various other properties that might matter, such as IsNotified, GuestCount, etc.)
4. others?

**Actual behavior:** 

A person that was an active member suddenly becomes inactive.

### Versions

* **Rock Version:** 6.8, 7.0
* **Client Culture Setting:** en-US

**Comment by edmistj** [comment]
I would say that in the case of both an Active and Inactive group member that the Active should take precedence.

**Comment by cabal95** [comment]
So, essentially make the priority logic for "overlapping" GroupMember records `Active > Pending > Inactive` ?

**Comment by cabal95** [comment]
@edmistj I can't find where the logic for GroupMember is in a person merge. I checked the PersonMerge block and a few other places and I just can't find it. I'm sure I'm missing it, because I also can't find where the PersonAlias is updated with the new Person Id. If one of you can point me in the right direction I can work on this.

**Comment by edmistj** [comment]
@cabal95 I believe it's in the stored proc spCrm_PersonMerge

**Comment by cabal95** [comment]
@edmistj I think I'm going to turn this one back over to you guys. Thanks for the point towards the sproc. I found it but am at a loss for how to go about doing this in SQL.

Additionally, while toying around with different ways to do this in SQL I found another problem (very related) to this. If one of the GroupMember records has notes on it and is the one chosen for deletion, those notes are lost. So I think this whole GroupMember decision making in merging records might need some overall TLC.
